packages/xo-web/src/common/utils.js

```diff
@@ -570,3 +570,21 @@ export const generateRandomId = () =>
   Math.random()
     .toString(36)
     .slice(2)
+
+// ===================================================================
+
+// Dedicated to Freactal
+// It only supports native inputs
+export const linkState = (_, { target }) => state => ({
+  ...state,
+  [target.name]:
+    target.nodeName.toLowerCase() === 'input' &&
+    target.type.toLowerCase() === 'checkbox'
+      ? target.checked
+      : target.value,
+})
+
+export const toggleState = (_, { target: { name } }) => state => ({
+  ...state,
+  [name]: !state[name],
+})
```

Comment:
Maybe create a dedicated `freactal-utils` module?

Also, add a link to the related Freactal issue and a todo to remove them once resolved.